tests/helpers.py

```diff
@@ -90,20 +90,41 @@ def assert_startswith(x: str):
 
 # Checks that a json blob (represented as a Python dict) is equal-ish to an expected
 # dict. The expected dict can contain assertion functions in place of any non-deterministic values.
-def compare_json(actual_json, expected_json):
+def compare_json(
+    actual_json, expected_json, keypath: Optional[List[Union[str, int]]] = None
```

Comment:
suggest calling the argument something like `current_keypath` or `keypath_breadcrumbs`, or something that indicates that you're just using it to keep track of the keypath so far.

Reply:
Renamed it `current_keypath`, moved the guts to an inner function, and added tests for `compare_json` specifically.